code/modules/events/aurora_locus.dm

```diff
@@ -0,0 +1,71 @@
+/datum/round_event_control/aurora_caelus
+	name = "Aurora Caelus"
+	typepath = /datum/round_event/aurora_caelus
+	max_occurrences = 1
+	weight = 15
+	earliest_start = 5 MINUTES
+
+/datum/round_event_control/aurora_caelus/canSpawnEvent(players, gamemode)
+	if(!CONFIG_GET(flag/starlight))
+		return
+	return ..()
+
+/datum/round_event/aurora_caelus
+	announceWhen = 1
+	startWhen = 10
+	endWhen = 50
+	var/list/aurora_colors = list("#A2FF80", "#A2FF8B", "#A2FF96", "#A2FFA5", "#A2FFB6", "#A2FFC7", "#A2FFDE")
+	var/aurora_progress = 0 //this cycles from 1 to 7, slowly changing colors from gentle green to gentle blue
+
+/datum/round_event/aurora_caelus/announce()
+	priority_announce("[station_name()]: A harmless cloud of ions is approaching your station, and will exhaust their energy battering the hull. \
+	Nanotrasen has approved a short break for all employees to relax and observe this very rare event. During this time, starlight will be bright \
+	but gentle, shifting between quiet green and blue colors. Any staff who would like to view these lights for themselves may proceed to the area \
+	nearest to them with viewing ports to open space. We hope you enjoy the lights.",
+	sound = 'sound/misc/notice2.ogg',
+	sender_override = "Nanotrasen Meteorology Division")
+	for(var/mob/M in GLOB.player_list)
+		if((M.client.prefs.toggles & SOUND_MIDI) && is_station_level(M.z))
+			M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
+
+/datum/round_event/aurora_caelus/start()
+	for(var/area in GLOB.sortedAreas)
+		var/area/A = area
+		if(initial(A.dynamic_lighting) == DYNAMIC_LIGHTING_IFSTARLIGHT)
+			for(var/turf/open/space/S in A)
```

Comment:
Potentially laggy

Reply:
I tested it locally, and had no issues. What would you recommend instead?